### PR TITLE
feat(test): gov/governance realm tests

### PR DESCRIPTION
**Pull request by jinoosss**
### Description

This PR enhances the governance realm testing infrastructure and fixes critical realm parameter handling issues across multiple modules.

#### Key Changes

**1. Enhanced Governance Parameter Handler Tests**
- Refactored `fn_registry_test.gno` with comprehensive test cases covering success and failure scenarios
- Added structured test data with explicit error expectations (panic, abort, error)
- Improved test readability with descriptive test names and better error handling
- Added test cases for invalid parameter validation across multiple modules

**2. Fixed Realm Parameter Handling**
- **Emission Module**: Updated `callback.gno` and `distribution.gno` to properly handle realm parameters
- **Governance Module**: Removed unnecessary realm parameter from `createParameterHandlers()` function
- **Distribution Functions**: Added realm parameter to `setDistributionBpsPct()` and related functions
- **Callback Mechanism**: Enhanced callback handling with proper realm context

**3. Improved Test Environment Management**
- Added `resetProposalStateEnv()` function for clean test state management
- Implemented proper test isolation with environment reset between tests
- Enhanced proposal state configuration management

#### Bug Fixes

- Fixed realm parameter inconsistency in governance parameter handlers
- Resolved callback execution context issues in emission distribution
- Corrected function signature mismatches in distribution percentage handling
- Fixed test state pollution issues in proposal state tests

#### Testing Improvements

- **Comprehensive Coverage**: Added tests for all major governance parameter handlers
- **Error Scenarios**: Included tests for invalid parameters, boundary conditions, and error cases
- **State Management**: Improved test isolation and cleanup procedures
- **Validation**: Enhanced parameter validation testing across multiple modules

#### Modules Affected

- `contract/r/gnoswap/emission/` - Callback and distribution handling
- `contract/r/gnoswap/gov/governance/` - Parameter handler tests and execution
- `contract/r/gnoswap/gov/governance/` - Proposal state management

#### Technical Details

- **Realm Context**: Proper realm parameter propagation throughout the governance system
- **Callback Safety**: Enhanced callback execution with proper realm context
- **Test Architecture**: Improved test structure with better error expectation handling
- **State Isolation**: Clean test environment management to prevent test interference

### CODE REVIEW

File: contract/r/gnoswap/emission/callback.gno [hunk 1]
```diff
@@ -17,7 +17,11 @@ func SetCallbackStakerEmissionChange(cur realm, callback func(amount int64)) {
 	if callbackStakerEmissionChange != nil {
 		panic("callbackStakerEmissionChange already set")
 	}
-	callbackStakerEmissionChange = callback
+	callbackStakerEmissionChange = func(amount int64) {
```
[the remark this answers is not in the document]
Comment (notJoon) [reply contract/r/gnoswap/emission/callback.gno:20]: It does the same behaviours. seems fine to me

**Review by notJoon — APPROVED** [review]
Greate improvement. LGTM
